lib/Conversion/TorchToTMTensor/TorchToTMTensor.cpp

```diff
@@ -523,6 +548,50 @@ class ConvertAtenMaxPool2dWithIndicesBackwardOp
 };
 } // namespace
 
+namespace {
+class ConvertAtenCumsumOp : public OpConversionPattern<AtenCumsumOp> {
+public:
+  using OpConversionPattern::OpConversionPattern;
+  LogicalResult
+  matchAndRewrite(AtenCumsumOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+
+    Value input = adaptor.self();
+    auto resultType = input.getType().cast<RankedTensorType>();
+    Type elementType = resultType.getElementType();
+    Location loc = op->getLoc();
+    SmallVector<Value> sizes = getTensorSizes(rewriter, loc, input);
+    Value output = createZeroInitTensor(rewriter, loc, sizes, elementType);
+    output = rewriter.create<tensor::CastOp>(loc, resultType, output);
+
+    int64_t dim;
+    if (!matchPattern(op.dim(), m_TorchConstantInt(&dim)))
+      return rewriter.notifyMatchFailure(
+          op, "unimplemented: only constant dim value is supported");
+
+    sizes.erase(sizes.begin() + dim);
+    SmallVector<int64_t> accStatic(resultType.getShape());
+    accStatic.erase(accStatic.begin() + dim);
+    Value acc = createZeroInitTensor(rewriter, loc, sizes, elementType);
+    Type accType = RankedTensorType::get(accStatic, elementType);
+    acc = rewriter.create<tensor::CastOp>(loc, accType, acc);
```

Comment:
Why are the `output` and `acc` casts needed?

Reply:
I was having issues with input being statically sized but `getZeroInitTensor` returning a dynamic tensor, and casting them all to input's sizes (with one less dim in the accumulator's case) was simpler than dealing with both static and dynamic input. If you think it'd be better practice/improve readability I can change that logic.

Reply:
I see. Yeah, this should be fine.